### PR TITLE
chore: deprecate TableViewRow header and footer properties

### PR DESCRIPTION
**Pull request by garymathews**
- Deprecate `TableViewRow` properties `header` and `footer` in favor of `headerTitle` and `footerTitle`

##### TEST CASE
- Both `headerTitle` and `footerTitle` should have priority over deprecated `header` and `footer` properties
- Using `header` and `footer` will output a deprecation warning
```JS
const win = Ti.UI.createWindow();
const tableView = Ti.UI.createTableView({ backgroundColor: 'white' });
tableView.setData([{
	headerTitle: 'headerTitle',
	footerTitle: 'footerTitle',
	header: 'header',
	footer: 'footer',
	title: 'Row',
	color: 'black'
}]);
win.add(tableView);
win.open();
```

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-28198)

### CODE REVIEW

**Review by jquick-axway — APPROVED** [review]
CR: Pass

File: iphone/Classes/TiUITableViewRowProxy.m [hunk 1]
```diff
@@ -314,18 +314,35 @@ - (void)updateRow:(NSDictionary *)data withObject:(NSDictionary *)properties
...
-  }
-  if ([data objectForKey:@"footer"]) {
-    [section setValue:[data objectForKey:@"footer"] forUndefinedKey:@"footerTitle"];
-    // we can return since we're reloading the section, will cause the
-    // row to be repainted at the same time
+  id headerTitle = [data objectForKey:@"headerTitle"];
+  if (headerTitle == nil) {
+    headerTitle = [data objectForKey:@"header"];
+
+    if (headerTitle != nil)
```
Comment (sgtcoolguy) [inline iphone/Classes/TiUITableViewRowProxy.m:321]:
```suggestion
    if (headerTitle != nil) {
```

File: iphone/Classes/TiUITableViewRowProxy.m [hunk 2]
```diff
@@ -314,18 +314,35 @@ - (void)updateRow:(NSDictionary *)data withObject:(NSDictionary *)properties
...
+
+  // Update section header with new headerTitle.
+  [section setValue:headerTitle forUndefinedKey:@"headerTitle"];
+}
+
+id footerTitle = [data objectForKey:@"footerTitle"];
+if (footerTitle == nil) {
+  footerTitle = [data objectForKey:@"footer"];
+
+  if (footerTitle != nil)
```
Comment (sgtcoolguy) [inline iphone/Classes/TiUITableViewRowProxy.m:335]:
```suggestion
  if (footerTitle != nil) {
```

**Review by sgtcoolguy — CHANGES_REQUESTED** [review]
iOS side doesn't compile

**Review by sgtcoolguy — APPROVED** [review]

**Comment by vijaysingh-axway** [comment]
@garymathews  In iOS, you have to make changes at couple of more places inside file TiUITableViewProxy.m e.g. https://github.com/appcelerator/titanium_mobile/blob/9b27f4ba0637d5fa98920b5262b5f2b7f8625782/iphone/Classes/TiUITableViewProxy.m#L694 , a=https://github.com/appcelerator/titanium_mobile/blob/9b27f4ba0637d5fa98920b5262b5f2b7f8625782/iphone/Classes/TiUITableViewProxy.m#L671 and https://github.com/appcelerator/titanium_mobile/blob/9b27f4ba0637d5fa98920b5262b5f2b7f8625782/iphone/Classes/TiUITableViewProxy.m#L603 , https://github.com/appcelerator/titanium_mobile/blob/9b27f4ba0637d5fa98920b5262b5f2b7f8625782/iphone/Classes/TiUITableViewProxy.m#L532

**Review by vijaysingh-axway — APPROVED** [review]
Looks good!

**Comment by ssekhri** [comment]
FR Passed.
Verified on:
Mac OS: 10.15.4
SDK: 10.1.0.v20210411201210
Appc CLI: 8.1.1
JDK: 11.0.6
Node: 12.16.1
Studio: 6.0.0.202005141803
Xcode: 12.3
iOS simulator v14.3
Android emulator v11.0

**Pull request merged**